### DISCUSSION TITLE
Not able to open an page, getting 404 not found

### DISCUSSION THREAD

**Discussion by naveensivakumar013** [cypress-io/cypress#27287] [Questions and Help]
Hi , Below is the scenario that I am trying to achieve, but that is not working when automating via cypress. But it is working fine wile performing manually.

Manual behaviour
- Opened an page directly for an example https://domain.com/admin
- It will ask for login and did logged in via sso
- It got redirected to the multiple page and land in admin page successfully

Cypress behavior
- Opened an page directly for an example https://domain.com/admin
- It will ask for login and did logged in via sso.
- login also happening properly after redirection
- While landing it to admin page it is saying 404 not found.

On the above steps, we are sure that it is not due to multiple redirection, because if we run the same script without admin, it will land into the home page without any error.

When we are analysing, we noticed that one difference from home page and admin page. In home page there is no Iframe, but in admin page, we noticed that entire page itself is in the iframe including source. We are not sure, whether that is the root cause for that issue. Can anyone help on this to resolve this issue?

**Comment by portyu9** [comment]
Hi @naveensivakumar013 

The iframe is likely the key difference.

First check whether the iframe `src` is **same-origin or cross-origin**.

* If it is **same-origin**, make sure the SSO session/cookies are established before `/admin` loads, then access the iframe through its `contentDocument`.
* If it is **cross-origin**, Cypress cannot directly interact with that iframe because of browser same-origin restrictions. `cy.origin()` does not work inside iframes.

The fix for a cross-origin iframe is to test the iframe application directly as a top-level page, or change the application/test setup so the embedded content is same-origin.

Also check the Network tab for the exact request returning `404`. If the iframe `src` itself returns `404`, verify that the SSO cookies/token are being sent with that request, because the problem is authentication/session handling rather than Cypress navigation.
